projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php

```diff
@@ -407,45 +407,68 @@ public function is_task_id_complete( $task_id ) {
 	 *
 	 * @param Task_List $task_list Task List.
 	 *
-	 * @return bool True if valid, false if not.
+	 * @return null|WP_Error Null if valid, WP_Error if not.
 	 */
 	public static function validate_task_list( $task_list ) {
+		$error_code     = 'validate_task_list';
+		$error_messages = array();
+
 		if ( ! is_array( $task_list ) ) {
-			return false;
+			// Ensure we have a valid task list array.
+			$msg = 'Invalid task list';
+			_doing_it_wrong( 'validate_task_list', esc_html( $msg ), '6.1' );
+			return new WP_Error( $error_code, $msg );
 		}
 
 		if ( ! isset( $task_list['id'] ) ) {
-			_doing_it_wrong( 'validate_task_list', 'The Launchpad task list being registered requires a "id" attribute', '6.1' );
-			return false;
+			// Ensure we have an id.
+			$msg = 'The Launchpad task list being registered requires a "id" attribute';
+			_doing_it_wrong( 'validate_task_list', esc_html( $msg ), '6.1' );
+			$error_messages[] = $msg;
 		}
 
 		if ( ! isset( $task_list['task_ids'] ) ) {
-			_doing_it_wrong( 'validate_task_list', 'The Launchpad task list being registered requires a "task_ids" attribute', '6.1' );
-			return false;
+			// Ensure we have task_ids.
+			$msg = 'The Launchpad task list being registered requires a "task_ids" attribute';
+			_doing_it_wrong( 'validate_task_list', esc_html( $msg ), '6.1' );
+			$error_messages[] = $msg;
+		} elseif ( isset( $task_list['required_task_ids'] ) ) {
+				// Ensure we have a valid array.
```

Comment:
Nit: the whitespace looks overly indented.
```suggestion
			// Ensure we have a valid array.
```